packages/protocol/contracts/common/FeeCurrencyWhitelist.sol

```diff
@@ -46,7 +53,51 @@ contract FeeCurrencyWhitelist is
    * @return Patch version of the contract.
    */
   function getVersionNumber() external pure returns (uint256, uint256, uint256, uint256) {
-    return (1, 1, 1, 0);
+    return (1, 1, 2, 0);
+  }
+
+  // removed with address zero
+  function setAdaptor(address token, address adaptor) external onlyOwner {
+    adapters[token] = adaptor;
+    emit AdaptorSet(token, adaptor);
+  }
+
+  // TODO make external
+  function getAdaptor(address underlyingToken) public view returns(address){
+    address result = adapters[underlyingToken];
+    return (result != address(0))? result: underlyingToken;
+  }
+
+  function setUnderlyinToken(address tokenAddress) external onlyOwner() {
+    underlyingTokens.push(tokenAddress);
+    emit UnderlyinTokenSet(tokenAddress);
+  }
+
+  function getUnderlyingTokens() external view returns(address[] memory){
+    return underlyingTokens;
+  }
+
+  // TODO when this contracts gets moved to Solidity 0.8 it should
+  // return an array of tuples
+  // secuence of underlyingtoken and it's corresponding adapter
+  function getWhitelistUnderlyngPairs() external view returns(address[] memory){
```

Comment:
Can we instead return 2 arrays ?

Reply:
it's not possible to have nested collections without experimental ABI on Solidity 0.5.*

Reply:
please check my PR - you can return two arrays

Reply:
what PR? you mean by returning (address[], address[])?